src/main/java/kyrene/command/Command.java

```diff
@@ -0,0 +1,244 @@
+                throw new KyreneMissingTaskException();
+            }
+            break;
+        case "deadline":
+            try {
+                tasks.add(new Deadline(description.substring("deadline ".length())));
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new KyreneMissingTaskException();
+            } catch (KyreneMissingTimeException e) {
+                Ui.showErrorMissingTime();
+                return;
+            }
+            break;
+        case "event":
+            try {
+                tasks.add(new Event(description.substring("event ".length())));
+            } catch (StringIndexOutOfBoundsException e) {
+                throw new KyreneMissingTaskException();
+            } catch (KyreneMissingTimeException e) {
+                Ui.showErrorMissingTime();
+                return;
+            }
+            break;
+        default:
+            throw new KyreneInvalidCommandException();
+        }
+
+        try {
+            storage.write(tasks);
+        } catch (IOException e) {
+            Ui.showErrorWriteToFileFailed();
+        }
+
+        int taskCount = tasks.size();
+        String taskAdded = tasks.get(taskCount - 1).toString();
+        Ui.showSuccessAddingTask(taskAdded, taskCount);
+    }
+
+    private void handleDelete(TaskList tasks, Storage storage) throws KyreneTaskNotFoundException {
+        if (taskNumber < 1 || taskNumber > tasks.size()) {
+            throw new KyreneTaskNotFoundException();
+        }
+        String taskDeleted = tasks.get(taskNumber - 1).toString();
+        tasks.remove(taskNumber - 1);
+
+        try {
+            storage.write(tasks);
+        } catch (IOException e) {
+            Ui.showErrorWriteToFileFailed();
+        } finally {
+            int taskCount = tasks.size();
+            Ui.showSuccessDeletingTask(taskDeleted, taskCount);
+        }
+    }
+
+    private void handleAT(TaskList tasks) {
+        LocalDate date = Parser.parseDate(description);
+        TaskList events = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskType().equals("E") && ((Event) task).isAt(date)){
+                events.add(task);
+            }
+        }
+        Ui.showEvents(events);
+    }
+
+    private void handleDue(TaskList tasks) {
+        LocalDateTime time = Parser.parseTime(description);
+        TaskList deadlines = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskType().equals("D") && ((Deadline) task).isBefore(time)){
+                deadlines.add(task);
+            }
+        }
+        Ui.showDeadlines(deadlines);
+    }
+
+    private void handleFind(TaskList tasks) {
+        TaskList matches = new TaskList();
+        for (Task task : tasks) {
+            if (task.getTaskName().contains(description)) {
+                matches.add(task);
+            }
+        }
+        Ui.showMatches(matches);
+    }
```

Comment:
These are good for modularity. However, consider refactoring them to other classes.